File: src/abstractFactory/java/factory/TeemoFactory.java
package abstractFactory.java.factory;

import abstractFactory.java.temmo.TeemoAbility;
import abstractFactory.java.temmo.TeemoConfig;

/**
 * Created by kvirus on 2019/6/7 19:44
 * Email @ caoyingkui@pku.edu.cn
 * <p>
 * |   *******    **      **     **     **
 * |  **            **  **       **  **
 * |  **              **         ***
 * |  **              **         **  **
 * |   *******        **         **     **
 */
public class TeemoFactory implements ChampionFactory {
    @Override
    public ChampionAbility getAbilities() {
        return new TeemoAbility();
    }

    @Override
    public ChampionConfig getConfig() {
        return new TeemoConfig();
    }
}

File: src/abstractFactory/kotlin/factory/TeemoFactory.kt
package abstractFactory.kotlin.factory

import abstractFactory.kotlin.teemo.TeemoAbility
import abstractFactory.kotlin.teemo.TeemoConfig

/**
 * Created by kvirus on 2019/6/7 20:25
 * Email @ caoyingkui@pku.edu.cn
 *
 * |   *******    **      **     **     **
 * |  **            **  **       **  **
 * |  **              **         ***
 * |  **              **         **  **
 * |   *******        **         **     **
 */
class TeemoFactory: ChampionFactory{
    override fun getAbilities(): ChampionAbility {
        return TeemoAbility()
    }

    override fun getConfig(): ChampionConfig {
        return TeemoConfig()
    }
}